language/stdlib/modules/transaction_fee.move

```diff
@@ -1,105 +1,87 @@
-    // the fees from this account from block metadata transactions.
-    public fun initialize_transaction_fees(fee_account: &signer) {
+    resource struct LBRIdent<LBRCoinType> { coin: Libra<LBRCoinType> }
+
+    // Called in genesis. Once for each fiat currency.
+    public fun add_transaction_fee<CoinType>(fee_account: &signer, preburn: Libra::Preburn<CoinType>) {
         Transaction::assert(Signer::address_of(fee_account) == 0xFEE, 0);
-        move_to(
-            fee_account,
-            TransactionFees {
-                fee_withdrawal_capability: LibraAccount::extract_sender_withdrawal_capability(),
-            }
-        );
+        move_to(fee_account, CollectedFees<CoinType> { preburn })
     }
 
-    // Called from block metadata transaction for each currency
-    public fun distribute_transaction_fees<Token>() acquires TransactionFees {
-      // Can only be invoked by LibraVM privilege.
-      Transaction::assert(Transaction::sender() == 0x0, 33);
-
-      let num_validators = LibraSystem::validator_set_size();
-      let amount_collected = LibraAccount::balance<Token>(0xFEE);
-
-      // If amount_collected == 0, this will also return early
-      if (amount_collected < num_validators) return ();
-
-      // Calculate the amount of money to be dispursed, along with the remainder.
-      let amount_to_distribute_per_validator = per_validator_distribution_amount(
-          amount_collected,
-          num_validators
-      );
-
-      // Iterate through the validators distributing fees equally
-      distribute_transaction_fees_internal<Token>(
-          amount_to_distribute_per_validator,
-          num_validators,
-      );
+    // Called in genesis.
+    public fun initialize(fee_account: &signer) {
+        Transaction::assert(Signer::address_of(fee_account) == 0xFEE, 0);
+        move_to(fee_account, LBRIdent<LBR>{ coin: Libra::zero<LBR>() });
     }
 
-    // After the book keeping has been performed, this then distributes the
-    // transaction fees equally to all validators with the exception that
-    // any remainder (in the case that the number of validators does not
-    // evenly divide the transaction fee pot) is distributed to the first
-    // validator.
-    fun distribute_transaction_fees_internal<Token>(
-        amount_to_distribute_per_validator: u64,
-        num_validators: u64,
-    ) acquires TransactionFees {
-        let distribution_resource = borrow_global<TransactionFees>(0xFEE);
-        let index = 0;
+    // Called for LibraAccount setup
+    public fun grant_fee_payment_capability(libra_account_singleton_addr: &signer): FeePaymentCapability {
+        Transaction::assert(Signer::address_of(libra_account_singleton_addr) == 0xA550C18, 0);
+        FeePaymentCapability {}
+    }
 
-        while (index < num_validators) {
+    public fun pay_fee<CoinType>(fee: Libra<CoinType>, _cap: &FeePaymentCapability)
```

Comment:
Wondering: do we need `FeePaymentCapability`? It doesn't seem particularly problematic if someone wants to send funds to the fee pot, and it's not possible without custom transaction scripts anyway.

Reply:
More than happy to take it away :) I don't think there should be adverse consequences, just that other people can pay into it with custom transaction scripts.